contracts/sfc/Staker.sol

```diff
@@ -609,6 +609,25 @@ contract Stakers is Ownable, StakersConstants {
         emit UnstashedRewards(auth, receiver, rewards);
     }
 
+    // stashed rewards are burnt on deactivation in all the cases except when delegator has deactivated after
+    // validator has deactivated or was slashed/pruned
+    function _rewardsBurnableOnDeactivation(bool isDelegation, uint256 stakerID) public view returns(bool) {
+        return !isDelegation || (stakers[stakerID].stakeAmount != 0 && stakers[stakerID].status == OK_STATUS && stakers[stakerID].deactivatedTime == 0);
+    }
+
+    // proportional part of stashed rewards are burnt on deactivation if _rewardsBurnableOnDeactivation returns true
+    function _mayBurnRewardsOnDeactivation(bool isDelegation, uint256 stakerID, address addr, uint256 withdrawAmount, uint256 totalAmount) internal {
```

Comment:
- perhaps can improve readability by removing isDelegation parameter in both methods _rewardsBurnableOnDeactivation and _mayBurnRewardsOnDeactivation.

=>
function _rewardsBurnableOnDeactivation(uint256 stakerID) public view returns(bool) {
        return (stakers[stakerID].stakeAmount != 0 && stakers[stakerID].status == OK_STATUS && stakers[stakerID].deactivatedTime == 0);
    }


function _burnRewardsOnDeactivation (bool isDelegation, uint256 stakerID, address addr, uint256 withdrawAmount, uint256 totalAmount) internal { 
   uint256 leftAmount = totalAmount.sub(withdrawAmount);
            uint256 newStash = rewardsStash[addr].amount.mul(leftAmount).div(totalAmount);
            if (newStash == 0) {
                delete rewardsStash[addr];
            } else {
                rewardsStash[addr].amount = newStash;
            }
}

Then line 641 (similarly line 666)
_mayBurnRewardsOnDeactivation(false, stakerID, stakerSfcAddr, stakers[stakerID].stakeAmount, stakers[stakerID].stakeAmount);
=>
// add comment: isDelegation is false
_burnRewardsOnDeactivation(stakerID, stakerSfcAddr, stakers[stakerID].stakeAmount, stakers[stakerID].stakeAmount);


Then line 725, 754:
_mayBurnRewardsOnDeactivation(true, stakerID, delegator, delegations[delegator].amount, delegations[delegator].amount);
=>
// add comment why isDelegation is true
if (_rewardsBurnableOnDeactivation(stakerID)) {
    _burnRewardsOnDeactivation(true, stakerID, delegator, delegations[delegator].amount, delegations[delegator].amount);
}